src/scripts/components/DropdownCheckbox.jsx

```diff
@@ -0,0 +1,268 @@
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import { Button, Checkbox, DropdownMenuItem, DropdownMenuItemWild, DropdownMenuScroll, DropdownFilter, DropdownWrapper, Icon } from '../components';
+
+class DropdownCheckbox extends React.Component {
+  state = {
+    activeKey: this.props.activeKey,
+    isOpen: false,
+    hasFilter: false,
+    isCheckbox: this.props.isCheckbox,
+    checked: this.props.isChecked,
+  };
+
+  componentWillMount() {
+    React.Children.map(this.props.children, (child) => {
+      if (!child) return;
+      if (child.type === DropdownFilter) {
+        this.setState({ hasFilter: true });
+      }
+    });
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (newProps.activeKey !== this.props.activeKey) {
+      this.setState({
+        activeKey: newProps.activeKey,
+      });
+    }
+    if (newProps.isChecked !== this.props.isChecked) {
```

Comment:
@abhicool957  Since **isChecked** contains a boolean value, we do not need to check it's presence here. If we do apply the check to verify if the value is present, it would only work when value will be true.